### PR TITLE
Author page fixes: #4215, #4121

### LINKED ISSUE #4121: Lists widget erroring and not rendering on non-author pages
Seems possible related to https://github.com/internetarchive/openlibrary/releases/tag/deploy-2020-11-17

`/opt/openlibrary/deploys/openlibrary/bc485c1/openlibrary/templates/lists/widget.html: error in processing template: AttributeError: 'NoneType' object has no attribute 'key' (falling back to default template)`

Sentry: 
### Evidence / Screenshot (if possible)
image

### Relevant url?
e.g. https://openlibrary.org/authors/OL550077A/Robert_E._Staub

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** FF82
- **Operating system?** Win10
- **Environment (prod/dev/local)?** prod

### Stakeholders
@mekarpeles

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #4215, closes #4121 
Fixes author pages. Necessary for hotfixing #4215. Tested on dev. treating as P0

### CODE REVIEW

File: openlibrary/plugins/worksearch/code.py [hunk 1]
```diff
@@ -642,28 +642,34 @@ def works_by_author(akey, sort='editions', page=1, rows=100, has_fulltext=False,
...
         solr_select += '&sort=first_publish_year+asc'
     elif sort.startswith('new'):
         solr_select += '&sort=first_publish_year+desc'
     elif sort.startswith('title'):
         solr_select += '&sort=title+asc'
-    solr_select += "&facet=true&facet.mincount=1&f.author_facet.facet.sort=count&f.publish_year.facet.limit=-1&facet.limit=25&" + '&'.join("facet.field=" + f for f in facet_fields)
+    _facets = '&'.join("facet.field=" + f for f in facet_fields)
+    solr_select += "&facet=true&facet.mincount=1&f.author_facet.facet.sort=count"
+    solr_select += "&f.publish_year.facet.limit=-1&facet.limit=25"
+    solr_select += "&%s" % _facets
```
Comment (cclauss) [inline openlibrary/plugins/worksearch/code.py:671]: Too error-prone.  Let's do this with a `params` dict and use `urllib.parse.urlencode(params)` or better yet move to `requests`.

**Comment by cclauss** [comment]
Re-running tests.

File: openlibrary/plugins/worksearch/code.py (hunk 1 above)
Comment (cdrini) [reply openlibrary/plugins/worksearch/code.py:675]: Fixed :+1:

**Comment by cdrini** [comment]
Re-assigned to you, @cclauss ; please deploy to staging, test that it works, and then merge :)

**Comment by cclauss** [comment]
What was the advantage to making params a list of two-tuples?  I can see the disadvantages but can not see any advantages.

**Comment by cdrini** [comment]
We can't use a dict because we need multiple `fq` params 👍

**Review by cclauss — APPROVED** [review]
Looks good to me on staging.

**Pull request merged**